src/Vault.sol

```diff
@@ -21,32 +21,33 @@ import {IVaultAuth} from "./interfaces/IVaultAuth.sol";
 
 contract Vault is IVault, ERC4626 {
     using SafeERC20 for IERC20;
-    
+
     // CONSTANTS
-    uint256 public constant WAD = 1e18;
+    uint256 private constant WAD = 1e18;
 
     // IMMUTABLES
-    IPool         public immutable POOL;
-    PoolInfoUtils public immutable INFO;
-    Buffer        public immutable BUFFER;
+    IPool         private immutable POOL;
+    PoolInfoUtils private immutable INFO;
+    Buffer        private immutable BUFFER;
+
     IVaultAuth    public immutable AUTH;
     uint8         public immutable assetDecimals;
     uint256       public immutable LP_DUST;
-    
+
     // STATE VARIABLES
-    uint256[]                   public buckets;
+    uint8 private bolt; // reentrancy lock: 0 = off, 1 = on
```

Comment:
With the changes to the modifier, you may be able to make this public again. Just flagging in case you haven't already tried this